client/src/components/organisms/RequestGroupForm.tsx

```diff
@@ -0,0 +1,351 @@
+              .map((requestType) => requestType.name as string)
+            : []
+        )
+      }
+    });
+  }
+
+  const getNameError = (name: string): string => {
+    if (name.length === 0) {
+      return "Please enter a group name";
+    }
+    if (name.length > 20) {
+      return "Group name cannot exceed 20 characters";
+    }
+    if (props.requestGroups.find((requestGroup) => requestGroup.name === name && requestGroup._id !== props.requestGroupId)) {
+      return "There is already a group with this name";
+    }
+
+    return "";
+  }
+
+  const getInputRequestTypeNameError = (inputRequestTypeName: string): string => {
+    if (requestTypeNames.find(requestTypeName => requestTypeName === inputRequestTypeName)) {
+      return "There is already a type with this name";
+    }
+    else if (inputRequestTypeName.length > 40) {
+      return "Type name cannot exceed 40 characters";
+    }
+    return "";
+  }
+
+  const getRequestTypeNamesError = (requestTypes: Array<string>): string => {
+    if (requestTypes.length === 0) {
+
+      return "Please create at least 1 type";
+    }
+    return "";
+  }
+
+  const getDescriptionError = (description: string) => {
+    if (!description) {
+      return "Please enter a description";
+    }
+
+    const editorState = EditorState.createWithContent(convertFromRaw(JSON.parse(description)))
+    if (!editorState.getCurrentContent().hasText()) {
+      return "Please enter a description";
+    }
+
+    return "";
+  }
+
+  const getImageError = (image: string) => {
+    if (image === "") {
+      return "Please select an image";
+    }
+    return "";
+  }
+
+  const onNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const newName: string = event.target.value;
+
+    setChangeMade(true);
+    setName(newName);
+    setNameError(getNameError(newName))
+  }
+
+  const onAddRequestType = (newRequestTypeName: string) => {
+    const error = getInputRequestTypeNameError(newRequestTypeName);
+
+    if (error === "") {
+      setRequestTypeNames(requestTypeNames.concat([newRequestTypeName]));
+    }
+    setChangeMade(true);
+    setRequestTypesError(error);
+  }
+
+  const onDeleteRequestType = (targetRequestTypeName: string) => {
+    const newRequestTypeNames = requestTypeNames.filter((requestTypeName) => requestTypeName !== targetRequestTypeName);
+
+    setChangeMade(true);
+    setRequestTypeNames(newRequestTypeNames);
+    setRequestTypesError(getRequestTypeNamesError(newRequestTypeNames))
+  }
+
+  const onInputRequestTypeNameChange = (requestTypeName: string) => {
+    setChangeMade(true);
+    setRequestTypesError(getInputRequestTypeNameError(requestTypeName))
+    return true;
+  }
+
+  const onDescriptionChange = (newDescription: string) => {
+    setDescription(newDescription)
+    setDescriptionError(getDescriptionError(description))
+  }
+
+  const onImageChange = (newImage: string) => {
+    setImageError(getImageError(newImage))
```

Comment:
This is being done everywhere: setError(getError(stuff))

Instead, change "getError" to "updateError" and do the setting of the error inside those methods, to simplify the "onChange" methods and keep error/change behaviour separate

Reply:
i can make the `updateError` functions, but i'll need to keep the `getError` functions for `onSubmit`

Reply:
Actually, i feel like the code is too cluttered if we use `updateError` since there are other places where we use `getError` and don't immediately use `setError`

Reply:
For the places (e.g. onAddRequestType) where we don't immediately use setError, there would be no change in behaviour if we did immediately use it. Instead of "if (error === "")" do "if (requestTypesError === "")" for example.

Reply:
if we combine setters and getters, then `onAddRequestType` will look like this:
```
const  updateInputRequestTypeNameError(newRequestTypeName) {
    setInputRequestTypeNameError(getInputRequestTypeNameError(newRequestTypeName))
}
const onAddRequestType = (newRequestTypeName: string) => {
    updateInputRequestTypeNameError(newRequestTypeName);

    if (inputRequestTypeNameError === "") {
      setRequestTypeNames(requestTypeNames.concat([newRequestTypeName]));
    }
    setChangeMade(true);
  }
```

Since `setInputRequestTypeNameError` is async, it doesnt immediately change `inputRequestTypeNameError`

So the statement ` if (inputRequestTypeNameError === "")` won't reflect the most recent change in newRequestTypeName